shared/src/commonMain/kotlin/com/bunbeauty/shared/domain/feature/cart/EditCartProductUseCase.kt

```diff
@@ -0,0 +1,43 @@
+package com.bunbeauty.shared.domain.feature.cart
+
+import com.bunbeauty.shared.data.repository.AdditionRepository
+import com.bunbeauty.shared.data.repository.CartProductAdditionRepository
+import com.bunbeauty.shared.domain.feature.addition.GetIsAdditionsAreEqualUseCase
+import com.bunbeauty.shared.domain.repo.CartProductRepo
+
+class EditCartProductUseCase(
+    private val cartProductRepo: CartProductRepo,
+    private val cartProductAdditionRepository: CartProductAdditionRepository,
+    private val additionRepository: AdditionRepository,
+    private val getIsAdditionsAreEqualUseCase: GetIsAdditionsAreEqualUseCase,
+) {
+    suspend operator fun invoke(
+        cartProductUuid: String,
+        additionUuidList: List<String>,
+    ) {
+        val cartProduct =
+            cartProductRepo.getCartProduct(cartProductUuid = cartProductUuid) ?: return
+
+        if (
+            getIsAdditionsAreEqualUseCase(
+                initialCartProduct = cartProduct,
+                additionUuidList = additionUuidList
+            )
+        ) {
+            return
+        }
+
+        cartProduct.cartProductAdditionList.forEach { cartProductAddition ->
+            cartProductAdditionRepository.delete(cartProductAdditionUuid = cartProductAddition.uuid)
+        }
+
+        additionUuidList.forEach { additionUuid ->
+            additionRepository.getAddition(uuid = additionUuid)?.let { addition ->
```

Comment:
Let's pass `additionList: List<Addition>` instead of `additionUuidList: List<String>` to not get them again